packages/contracts-ops/contracts/CallBatch.sol

```diff
@@ -98,14 +101,34 @@ abstract contract CallBatch is Script {
     function finish() public {
         require(bytes(domain).length != 0, "must initialize");
         require(bytes(outputFile).length != 0, "must initialize");
+        require(!complete, "already written");
         complete = true;
         writeOutput();
     }
+
+    function build(address governanceRouter) public {
+        require(bytes(domain).length != 0, "must initialize");
+        require(bytes(outputFile).length != 0, "must initialize");
+        require(!complete, "already written");
+        complete = true;
```

Comment:
hmmm so you can only call `finish` or `build` - not both?
also - should finish be renamed something like `output` or `write`?

Reply:
why don't we just combine the functionalities of `finish` and `build` - writing the un-built and the built calldata to a file each time, similar to the typescript?

Reply:
because you'll never want both. either you want to combine the list with other lists and THEN build for governance (the `finish()` call), or you want to `build()` now for recovery

Reply:
mm when we used the typescript library, i often used the ability to look at the broken down calls in addition to the built calldata. to sanity check that the correct number of calls were there, etc.

Reply:
we do that because TS is junk and we want to know we wrote it write. much lower chance of error doing it in solidity